soc/arm64/nxp_imx/mimx9/mmu_regions.c

```diff
@@ -30,21 +30,15 @@ static const struct arm_mmu_region mmu_regions[] = {
 			      DT_REG_SIZE(DT_NODELABEL(ana_pll)),
 			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
 
-	MMU_REGION_FLAT_ENTRY("UART1",
-			      DT_REG_ADDR(DT_NODELABEL(lpuart1)),
-			      DT_REG_SIZE(DT_NODELABEL(lpuart1)),
-			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
-
-	MMU_REGION_FLAT_ENTRY("UART2",
-			      DT_REG_ADDR(DT_NODELABEL(lpuart2)),
-			      DT_REG_SIZE(DT_NODELABEL(lpuart2)),
-			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
-
 	MMU_REGION_FLAT_ENTRY("IOMUXC",
 			      DT_REG_ADDR(DT_NODELABEL(iomuxc)),
 			      DT_REG_SIZE(DT_NODELABEL(iomuxc)),
 			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
 
+	DT_FOREACH_STATUS_OKAY_VARGS(nxp_kinetis_lpuart,
+			  MMU_REGION_DT_FLAT_ENTRY,
+			 (MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS))
```

Comment:
`MMU_REGION_DT_FLAT_ENTRY`

Reply:
`MMU_REGION_DT_FLAT_ENTRY` is meant to be used within `DT_FOREACH_STATUS_OKAY_VARGS`.

Reply:
mmmh, right, can it be the opposite? Adding a `MMU_REGION_DT_FLAT_ENTRY` encapsulating the `DT_FOREACH_STATUS_OKAY_VARGS` ? That would be symmetrical to the current `MMU_REGION_FLAT_ENTRY` macro we have.

Reply:
It can be wrapped with another layer of macro. It will looks something like 

```c
MMU_REGION_DT_FLAT_ENTRY_STATUS_OKAY(compatiable, attr)
```

It looks really weird, AFAIK there aren't many macros that takes a `compatiable string` as a parameter.

ps: Please give me a better name lol

Reply:
Nah, using a compatible here looks fine to me. But I would call it `MMU_REGION_DT_FLAT_ENTRY(compatible, attr)` and rename the current macro to something internal.

Reply:
most macro with "SUBSYS"_DT_"DO_SOMETHING" macro takes node_id as first parameter. I'm unwilling to break the convention here...

Found this:

```c
#define DT_COMPAT_GET_ANY_STATUS_OKAY(compat)
```

so let's call it `MMU_REGION_DT_COMPAT_FLAT_ENTRY(compat, attr)` or `MMU_REGION_DT_COMPAT_FOREACH_STATUS_OKAY_FLAT_ENTRY(compat, attr)` (...it best describes the purpose, but too loooooong).